qiskit_machine_learning/neural_networks/estimator_qnn.py

```diff
@@ -85,7 +105,7 @@ class EstimatorQNN(NeuralNetwork):
     def __init__(
         self,
         *,
-        circuit: QuantumCircuit,
+        circuit: QuantumCircuit | QNNCircuit,
```

Comment:
Since a QNNCircuit is a QuantumCircuit its already really covered by the QuantumCircuit typehint. Yes we recognize this specific subclass and do things differently...